azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py

```diff
@@ -99,7 +107,7 @@ def on_connected():
             # See "A note on exception handling" above
             self.provider.on_mqtt_connected = on_connected
             try:
-                self.provider.reconnect(self.sas_token)
+                self.provider.connect(self.sas_token, self.certificate)
```

Comment:
Optional argument needs explicit keyword in call

Reply:
1) Why did we switch to `.connect()` instead of `.reconnect()`? Are you suggesting we deprecate `.reconnect()`?
2) Still not using kwargs

---
In reply to: [291739396](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291739396) [](ancestors = 291739396)

Reply:
it was a typo to do connect. reconnect is correct and the original code. for the reconnect the argument is not optional as defined in mqtt provider. 

---
In reply to: [291843682](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291843682) [](ancestors = 291843682,291739396)